### ISSUE TITLE
UI Upgrade for Address Selector

### ISSUE THREAD

**Issue by zakhap** [hicommonwealth/commonwealth#4315]
## Objective #2: Allowing users to easily switch between their different addresses with an improved universal top nav.

<img width="700" alt="Screen Shot 2023-07-06 at 1 11 54 PM" src="https://github.com/hicommonwealth/commonwealth/assets/16248081/701f545f-fc76-4f43-a992-a74af2b19db3">

### User Story 1 => Update Address selector dropdown.
- As a user, I need to be able to easily view my linked addresses along with their status (session key, “signed out”) and switch between them as I navigate Common, so I know the addresses and social logins I have linked to take the correct actions when I’m in specific communities. 
**Solution to implement:**
- New user dropdown using the new design systems dropdown component
  - User dropdown button updated
  - Dropdown to include two sections with sub-headers and a divider in-between each
  - Section 1 is `Addresses` with ~all the linked addresses~ active addresses and icons to show their chain (`icon` + `address`) or social login method (icon + socialUsername). => Update: the scope reduction from Objective 1 remains true here.
  - Section 2 is Settings which will include: (1) View profile, (2) Edit profile), (3) Notifications, and (4) Dark mode which includes a toggle => These are already existing dropdown menu buttons and will keep their functionality.

### User Story 2 => Update Universal Nav IconButtons
- As a user, I need to be able to clearly understand the universal actions I can take that is visually consistent with the rest of the product, so I can (1) view my notifications, (2) provide feedback, (3) explore help docs, and (4) explore other communities. 

**Solution to implement:**
- Use new icons from Phosphor
- **Explore** = `compass` / regular, `neutral-500`
- **Help** = `question` / regular, `neutral-500`
- **Send Feedback** = `PaperPlaneTilt` / regular, `neutral-500`
- **Notifications Bell** = `BellSimple` / regular, `neutral-500`.
- All above Icons are used as Icon Buttons.
  - All icons are 16x16px within a container with a 4px padding all around and a 6px radius for hover states. Each component will show up as 24x24px with the added padding and each icon button is separated by a spacing of 8px.
  - Hover state: The icon becomes Neutrals/600 with a background (entire container with the 6px radius) of Neutrals/100

### User Story 3 => Updating Logout button
- As I a user, I need to be able to log out of my Common account, so I have control and agency over how and when I use the platform. 
**Solution to implement:**
- `Logout` = `SignOut` / Regular, Neutrals/500 
  - (Behavior) Hover state: The icon becomes Neutrals/600 with a background (entire container with the 6px radius) of Neutrals/100

Out of Scope: 
- Search Bar
-

**Comment by masvelio** [comment]
@zakhap @meeshlin 

### User Story 1 Questions:
1. We do not have new dropdown component from the design system. Do we want to (1) implement it before this ticket and use new component or (2) use component we already have that looks very similar to the design but won't be 100% identical. 

**CURRENT COMPONENT THAT WE CAN USE AND SWAP LATER**
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/52c10380-ae3e-4418-8f9a-fd894db144fe)

**DESIGNED COMPONENT**
<img width="212" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/394006e8-c0d2-4db1-9148-63c6663b41bd">

2. [As discussed in comment (point 3)](https://github.com/hicommonwealth/commonwealth/issues/4413#issuecomment-1634493224) in other ticket about the addresses' icons: we are gonna use community avatars instead of designed icons. But does it also work for let's say social logins? If I login with google or github, should we show relevant social icon or the icon of the community that address belongs to?
3. Are those addresses that are `signed out` are clickable? What happens when I click it? Tbh I don't understand why they are `signed out` but this is probably something I can catch up with Ray or Jake.
4. I can't see `connect a new address` option in the dropdown any longer. Do we remove this functionality, it is moved somewhere else or it is oversight?
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/23a2cd4c-2522-4a88-9932-684de7ff5e94)
5. Currently in the dropdown we have option `Notification settings` but in the design we have `Notifications`. Former seems more accurate because it takes user to notification settings page, while bell icon is responsible for navigation to notifications page. 
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/7f1fc498-3e52-43b3-8543-a42492a3901b)

**Comment by masvelio** [comment]
@zakhap @meeshlin 

### User Story 2 Questions:
1. What about the plus icon that we currently have in the top nav bar. It is pretty important (although duplicated because it is also visible in left sidebar)
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/a41aeab9-42b1-4f7f-95c1-97b0db7cca49)
2. New designs looks great but I would love to hear from you what is your strategy for the mobile navigation? If we are modifying big-screen-nav, we probably should think about redesigning mobile-nav as well. Currently, in mobile dropdowns are the same as on desktop but the icons are hidden also in 3 dots menu that appears as a full width box. Please let me know how to approach mobile navigation in terms of this ticket. I would say that we have to rethink how the sidebar works in the future as well (it is related to the top nav especially on the mobile).

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/6c06a54a-4bdf-442f-b1d9-8dd98b50c82f)

**Comment by masvelio** [comment]
No questions to User Story 3

**Comment by zakhap** [comment]
> 
> ### User Story 1 Questions:
> 1. We do not have new dropdown component from the design system. Do we want to (1) implement it before this ticket and use new component or (2) use component we already have that looks very similar to the design but won't be 100% identical.
> 

- Aw shucks, I made the unfortunate assumption that because we use newly design dropdown components for sorting/filter threads and comments that they had been added as their own component. Do you mind providing how long you think it would take to generalize/componentize these components? 

> 2. [As discussed in comment (point 3)](https://github.com/hicommonwealth/commonwealth/issues/4413#issuecomment-1634493224) in other ticket about the addresses' icons: we are gonna use community avatars instead of designed icons. But does it also work for let's say social logins? If I login with google or github, should we show relevant social icon or the icon of the community that address belongs to?

- The behavior of the dropdown should be to only enable addresses that are able to join the community.
- The list of addresses shown are **not** _every address as role in a community_ (with community avatar) but _every unique address_ (in standard encodings: hex, bech32, etc). We should probably arrive at what chain icon to show by encoding type. @jnaviask to confirm. 
- For socials, we don't currently do "overrides" where we hide the address we generate using Magic and visually replace them with the socials used to generate those addresses. Unfortunately, it seems like we _only_ store "magic" as the wallet_id and not anything else about the social for overriding. @jnaviask is there something we can do here to do the overrides? If not, let's document that here and come up with a solution to show the addresses magic generated as such.

> 3. Are those addresses that are `signed out` are clickable? What happens when I click it? Tbh I don't understand why they are `signed out` but this is probably something I can catch up with Ray or Jake.

- "Signed Out" addresses are addresses that don't currently have an active session key. @meeshlin What do you think of replacing the copy of "Signed out" with "Inactive" (as that's more accurate).

> 4. I can't see `connect a new address` option in the dropdown any longer. Do we remove this functionality, it is moved somewhere else or it is oversight?

- We should absolutely maintain a "Connect a New Address" option at the bottom of the address selection options above the Settings section.

> 5. Currently in the dropdown we have option `Notification settings` but in the design we have `Notifications`. Former seems more accurate because it takes user to notification settings page, while bell icon is responsible for navigation to notifications page.

- "Notification settings" just is moved further down in the new popover menu under Settings as simply "Notifications". 
<img width="206" alt="Screen Shot 2023-07-14 at 12 26 52 PM" src="https://github.com/hicommonwealth/commonwealth/assets/16248081/d4807ff2-eddd-447a-8982-df246eededf1">

**Comment by zakhap** [comment]
> ### User Story 2 Questions:
> 1. What about the plus icon that we currently have in the top nav bar. It is pretty important (although duplicated because it is also visible in left sidebar)

- Ah superb question. I think we should keep it (if that's ok @meeshlin) and simply use the phosphor icon `plus-circle` for consistency.

> 2. New designs looks great but I would love to hear from you what is your strategy for the mobile navigation? If we are modifying big-screen-nav, we probably should think about redesigning mobile-nav as well. Currently, in mobile dropdowns are the same as on desktop but the icons are hidden also in 3 dots menu that appears as a full width box. Please let me know how to approach mobile navigation in terms of this ticket. 

- @meeshlin How would you like to handle mobile? In my opinion, we should simply have an option in the mobile menu for each of the icons that are hidden. Explore being the only new option there, it could easy just be added to the list of mobile options.

> I would say that we have to rethink how the sidebar works in the future as well (it is related to the top nav especially on the mobile).

- Coming soon 👀

**Comment by masvelio** [comment]
> > ### User Story 1 Questions:
> > 
> > 1. We do not have new dropdown component from the design system. Do we want to (1) implement it before this ticket and use new component or (2) use component we already have that looks very similar to the design but won't be 100% identical.
> 
> * Aw shucks, I made the unfortunate assumption that because we use newly design dropdown components for sorting/filter threads and comments that they had been added as their own component. Do you mind providing how long you think it would take to generalize/componentize these components?

@zakhap As I see [on the design](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=1%3A49&mode=design&t=HkJFg2dhfOP3dIsk-1) this is quite complicated component, with all those states, list placements, multiple types of elements inside etc. I would say that this is 2-3 days depending how much we can use from existing components (plus icon - create dropdown and sort/filter dropdowns -- these are two different components as well so it would need to be unified properly). Probably one day more to make it work in storybook. No matter what will be your decision, we need ticket for this. But let's decide in next 1-2 days which way we go!

**Comment by jnaviask** [comment]
@zakhap @masvelio @meeshlin 
> We should probably arrive at what chain icon to show by encoding type
Why not show the icon of the wallet used to link it?

> is there something we can do here to [show social login info]
Yes, but we will need to add it to the login flow. Magic exposes the following route:

```
GET https://auth.magic.link/v1/oauth2/user/info/retrieve
QUERY
  * provider     (string) - The name of the OAuth provider (i.e.: google,
                            facebook, twitter, apple).
  * field_format   (enum) - Option to return the user info fields as camelCase
                            or snake_case (one of: 'camelCase' | 'snake_case').
HEADERS
  * x-magic-api-key  (string) - The publishable API key for your Magic app.
  * authorization    (string) - An OAuth access token for the specified provider
                                in all cases except Apple, for which you should
                                provide a Magic DID token instead. Please
                                provide in a "Bearer {token}" format.
RESPONSE
  200/3xx:
    {
      data: {
        ...Open ID profile fields
        sources?: If applicable, an object mapping provider-specific endpoints
                  to their raw return value, we combine the result of these
                  endpoints to formulate the Open ID userinfo fields in the
                  `data` object.
      },
      error_code: "",
      message: "",
      status: "ok",
    }
  4xx:
    {
      data: {
        oauth: {
          provider: The OAuth provider (i.e.: google, facebook, twitter, apple).
          error: The OAuth2 error code.
          error_description?: The OAuth2 error description.
          error_uri?: The OAuth2 error URI.
        }
      },
      error_code: Same as `data.oauth.error`,
      message: A formatted error message containing information from
               `...data.oauth`.
      status: "failed",
    }
  5xx: Uh oh, get in touch with us because something's up!
```

We can use this route to query OAuth data at social login validation time, which would allow us to store the metadata about which method was used (and also "verified" social information such as usernames on other profiles).

However, this would be a fairly significant change (although we could start by simply tacking it onto the route), and does not yet have a ticket.

**Comment by masvelio** [comment]
Here I want to exclusively talk about icons that are part of **User Story 1**. 
- First of all I need you guys to agree on one solution because @zakhap is suggesting to "use the community icons that we [currently use next to the addresses on the editProfile](https://github.com/hicommonwealth/commonwealth/issues/4413#issuecomment-1634493224) page". @meeshlin wants to use custom chain icons defined in [figma](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=457%3A4520&mode=dev).
- If we choose to go with Zak's solution then this probably be simple (as it is already used in the code). If we choose to go with Meesh's proposition, I have some questions here.
  -  I would say again that we should have a defined list of supported icons. I was looking at the code and we can use `Address.chain.base` for `cosmos`, `substrate`, `ethereum`, `near`, `solana`. In this case we are missing `substrate` and `solana` icons @meeshlin. Correct me if I am wrong @zakhap but `Polygon` and `Polkadot` are not needed?
  - Another set of icons are socials. Currently in the code I can't see how we could identify if this address if from `discord`, `twitter` or `github`. All of them have property`walletId: magic`. Same for email/google authentication. @jnaviask  provided some suggestions how we can handle it so the question would be now @zakhap what should be the scope of the v1 and if we want to do any FF? 

For now I am gonna work on UI components so then when we will have the decisions, swapping proper icons will be easy.

**Comment by meeshlin** [comment]
@masvelio @zakhap 

### User Story 1

All icons can be found here: https://www.figma.com/file/yCUKk46hg9Fy3EXGkSoaKT/%F0%9F%9A%A7-Foundation?type=design&node-id=709%3A120&mode=dev

Do you need me to export them all in SVGs?

### User story 2

> @meeshlin How would you like to handle mobile? In my opinion, we should simply have an option in the mobile menu for each of the icons that are hidden. Explore being the only new option there, it could easy just be added to the list of mobile options.

For **native mobile** apps, you want to avoid dropdowns as much as possible—there are many solutions to this (ex. bottom sheet). Native apps use different patterns entirely for modals, dropdowns, and more. This is why you can't "rip and replace" an mWeb responsive experience to native mobile. However, in social apps, these navigational elements are exposed as a side nav. Refer to how this is designed in Reddit. You tap the user avatar and it exposes a right slide-out component. 

![IMG_0394](https://github.com/hicommonwealth/commonwealth/assets/18157060/8aae60d8-d51d-4dd5-a333-c76cbd4100a3)

**Comment by zakhap** [comment]
Being that we do not currently collect the wallet for addresses (re: @jnaviask's comment), we should probably default to the community icons for the moment to unblock this work.

As we continue to work toward clearing up our domain model of identity (addresses, login, etc), we can continue the work there with collecting the wallet and showing the wallet icons. This will be later work.

As for mobile menus, that is out of scope for this ticket.

**Comment by jnaviask** [comment]
Can we make a follow-up ticket to add icons, once #4574 is implemented? @zakhap

**Comment by masvelio** [comment]
I will start with work on UI component first and make the PR out. Then will handle the logic hookup

**Comment by masvelio** [comment]
@zakhap 
1. Asked already but did not get an answer. Can those items that are marked as `Signed out` be clickable? If yes, what happens if I click it? 
<img width="605" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/57c309df-f9d3-49ec-a005-8e28c0b73346">

2. Currently we have a case that if you do not have address that is based on the community you are watching, you can see the `NO ${chainName} Address` instead of the join button, which opens up login modal on click. I can't see something similar in the new designs. What we want to do about this scenario?

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/fdae986d-65fe-4984-96eb-d59132057528)

3. Is there any example in the figma, how icon button should look like with the default/hover/click state? The description is not super clear for me, especially that the icons on the designs have size of 24px, not 16 as described. 

<img width="444" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/6cab636d-8c82-42a6-9e06-edde2d65eed4">

> All icons are 16x16px within a container with a 4px padding all around and a 6px radius for hover states. Each component will show up as 24x24px with the added padding and each icon button is separated by a spacing of 8px.
> Hover state: The icon becomes Neutrals/600 with a background (entire container with the 6px radius) of Neutrals/100

**Comment by masvelio** [comment]
@meeshlin 

The design of the top navbar is missing the state when the user is logged out. Currently, it looks like this 

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/03138dca-b29e-4c9e-a07c-ac92f43aa8a7)

I am talking here specifically about the `Person & Log In` button in right hand side. Should this be unchanged or you want to update the icon/copy here?

**Comment by meeshlin** [comment]
If a user is completely logged out, the name component along with the `Log out` icon button will be replaced with a small, primary button that says `Login`. 

![00 Community - Not logged in](https://github.com/hicommonwealth/commonwealth/assets/18157060/a0ba5acc-2277-48bc-985e-6acfe71db106)

- [Figma prototype](https://www.figma.com/proto/mJMPTzEnpEFIaR2wBWkElq/Session-Keys-and-Login?page-id=516%3A8575&type=design&node-id=520-38622&viewport=-1404%2C600%2C0.89&t=Dhe60Oycn2jcosAT-1&scaling=scale-down&starting-point-node-id=520%3A38622&mode=design)
- [Figma dev mode](https://www.figma.com/file/mJMPTzEnpEFIaR2wBWkElq/Session-Keys-and-Login?type=design&node-id=516%3A8575&mode=dev)

@masvelio

**Comment by masvelio** [comment]
According to our latest agreements with @jnaviask and @zakhap, the current architecture of the app does not allow us achieve all desired behaviour prepared by the design team at this moment. 

**User Story 1** 
- user dropdown with addresses will work similarly as now, meaning, that we will display only `activeAccounts` (those addresses that have joined this particular community) instead of displaying all connected addresses as in the designs
- this work is ready on `raymond.react-session-keys-on`

**User Story 2 & 3** 
- no problems here, everything is ready on `raymond.react-session-keys-on`

@jnaviask @CowMuon considering this ☝️  I am wondering if we can close this ticket to make a `session-keys` project a bit more cleaner?
